### ISSUE TITLE
Option to change default async timeout

### ISSUE THREAD

**Issue by mjbeauregard** [Quick/Nimble#255]
I have a project that extensively uses `toEventually()` for async tests. These tests run great locally, but when TravisCI runs them, we often get random timeouts that seem to be solvable by manually increasing the timeout of each test with something like `toEventually(equal(1), timeout: 5)`.

Unfortunately, adding custom timeouts like this is sort of like whack-a-mole and is quickly leading to setting a custom timeout on every/most async test depending on how slow TravisCI is running during any build.

I'm wondering if this is a common problem for others or if there is an existing solution. If this **is** a common unsolved issue, I'd consider working on and submitting a PR that adds the feature (provided the maintainers are warm to the idea).

**Comment by jeffh** [comment]
Hey @mjbeauregard,

This is the first time someone has filed an issue to raise this issue. I personally prefer to avoid async tests when possible because it's time-dependent behavior.

That being said, I don't mind if there was a way to globally increase the timeout. Seems like a useful feature to have in tweaking the async timeout for CI.

**Comment by mjbeauregard** [comment]
I also much prefer to avoid async tests, but sometimes the things you are testing are inherently asynchronous. For example, when testing classes that use promises, even if the promise is fulfilled immediately you still need to do async testing due to the bit of thread-hopping inherent to that design. I don't think I can do much about that part. 

I wonder if this issue is related to this being a pretty large project (takes TravisCI over 20 minutes just to compile). As I mentioned, this problem never happens when running tests locally. I'll take a look and will probably submit a PR with the proposed change.

Thanks for your work on this fantastic library.

**Comment by jeffh** [comment]
I think it isn't too farfetched to see file an issue to the Promises library you're using. I haven't used much Promises libraries in Swift, but the one I used in Objective-C did support a blocking synchronous flow to allow tests to not be async (using semaphores).

That being said, see my comment in the PR, you might want to debug how much async work you have enqueued in your run loop.

**Comment by mjbeauregard** [comment]
I made an interesting discovery yesterday while debugging this. I noticed that our Travis config was explicitly running tests on the "iPad Air2" simulator (where we were getting timeouts), but I always test locally on the "iPad 2" simulator (where we never had any timeouts). 

I have verified by playing with the destination simulator setting that we are getting consistent results between Travis and local builds now. Both builds have timeouts when running on the "iPad Air2" simulator and and both pass on the "iPad 2".

I'm not sure why this would be the case just yet, but will keep digging and report back any findings. I agree that this could possibly be somehow related to the [SwiftTask](https://github.com/ReactKit/SwiftTask) library that we are using.
